### PR TITLE
feat: add no-duplicate-store-ids

### LINKED ISSUE #5: Rule: Unique store ids
See https://github.com/vuejs/pinia/issues/1394

### PR DESCRIPTION
**Pull request by 9romise**
# Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What did you change?

Add no-duplicate-store-ids, closes #5 

## Why did you change it?

This is a new rule

## Other information

I'm intersted on it and willing to contribute. Here is a commit with my rough idea. I'm not sure if there's a better one. If you have any suggestions, I'm looking forward to hearing them.

### CODE REVIEW

File: src/index.ts
Comment (lisilinhart) [inline src/index.ts:36]: I've been sticking to the convention to keep to **warnings** if **all** rules are enabled and only do the error warning on the 'recommended' rules. People can still set them to 'error' through the config. 

```suggestion
  [noDuplicateStoreIdsName]: 'warn'
```

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [inline src/rules/no-duplicate-store-ids.ts:8]: We can probably move this constant inside of the rule context

```suggestion
```

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [inline src/rules/no-duplicate-store-ids.ts:23]:
```suggestion
  create: (context) => {
  const usingStoreIds = new Set<string>()
```

**Review by lisilinhart — CHANGES_REQUESTED** [review]
Thanks for your contribution, I have some minor feedback, then we can merge it 🙂

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [reply src/rules/no-duplicate-store-ids.ts:23]: I think the rule should still work if the context of the store IDS is inside here :)

Comment (lisilinhart) [reply src/rules/no-duplicate-store-ids.ts:23]: Hi @9romise, sorry for the late reply. I've been travelling for work. I'm not 100% sure eslint rules should work across files, so I need to do some more tests on this in the next days before I merge it.

Comment (9romise) [reply src/rules/no-duplicate-store-ids.ts:23]: No worries at all! 😊 Thanks for letting me know. It's important to ensure the eslint rules are effective across files. I completely understand your decision to hold off on merging until further testing. Let me know if there's anything I can assist with in the meantime.

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [inline src/rules/no-duplicate-store-ids.ts:19]: Let's add more context here:

```suggestion
      duplicatedStoreIds: 'No duplicated store ids allowed: {{storeId}}'
```

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [inline src/rules/no-duplicate-store-ids.ts:38]: We can report on the actual literal node to make it more clear.

```suggestion
            node: storeId,
```

File: src/rules/no-duplicate-store-ids.ts
Comment (lisilinhart) [inline src/rules/no-duplicate-store-ids.ts:39]: And add the context here:

```suggestion
            messageId: 'duplicatedStoreIds',
            data: {
              storeId: storeId.value
            }
```

**Review by lisilinhart — COMMENTED** [review]
Hi @9romise, I tested your change here locally and you're right that it can only report across the files with the Set outside the eslint context. I'm not 100% if this might cause side effects, but it seems to work, so we can try adding it for now. I think we can just add a little bit more context and report closer to where the actual node is:

image

**Comment by 9romise** [comment]
Hi @lisilinhart, I think your suggestion is correct, I've completed these changes.

**Review by lisilinhart — APPROVED** [review]

**Pull request merged**
